### PR TITLE
airflow-provider-vdk: VDKSensor initial structure

### PR DESCRIPTION
**Pull request by gageorgiev**
This change lays out the initial structure for the VDK asynchronous
operator, a.k.a. the VDKSensor. The purpose of this sensor is to
routinely poke the VDK control-service and check the status of a
particular job, so that its completion/failure can trigger some
activity in an Airflow DAG, i.e. trigger another job.

Signed-off-by: Gabriel Georgiev <gageorgiev@vmware.com>

### CODE REVIEW

File: projects/vdk-plugins/airflow-provider-vdk/tests/sensors/test_vdksensor.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from unittest import mock
+
+from vdk_provider.sensors.vdk import VDKSensor
+
+
+@mock.patch.dict(
+    "os.environ", AIRFLOW_CONN_TEST_CONN_ID="http://https%3A%2F%2Fwww.vdk-endpoint.org"
```
Comment (doks5) [inline projects/vdk-plugins/airflow-provider-vdk/tests/sensors/test_vdksensor.py:9]: Should this be a URL?

**Review by doks5 — CHANGES_REQUESTED** [review]
Maybe add some docstrings about what the sensor is supposed to do. No need to be detailed (as it would be improved during the implementation process), just some high-level information.

File: projects/vdk-plugins/airflow-provider-vdk/tests/sensors/test_vdksensor.py (hunk 1 above)
Comment (gageorgiev) [reply projects/vdk-plugins/airflow-provider-vdk/tests/sensors/test_vdksensor.py:9]: It works for the hook tests - I copied the format over from the Airflow docs.

Comment (doks5) [reply projects/vdk-plugins/airflow-provider-vdk/tests/sensors/test_vdksensor.py:9]: Mind sharing a link?

Comment (gageorgiev) [reply projects/vdk-plugins/airflow-provider-vdk/tests/sensors/test_vdksensor.py:9]: I was wrong, it's not the docs, it's here - https://github.com/astronomer/airflow-provider-sample/blob/main/tests/hooks/test_sample_hook.py#L27

File: projects/vdk-plugins/airflow-provider-vdk/vdk_provider/sensors/vdk.py [hunk 2]
```diff
@@ -1,2 +1,29 @@
...
 # SPDX-License-Identifier: Apache-2.0
+from typing import Any
+from typing import Dict
+
+from airflow.sensors.base import BaseSensorOperator
+from vdk_provider.hooks.vdk import VDKHook
+
+
+class VDKSensor(BaseSensorOperator):
+    def __init__(
```
Comment (mivanov1988) [inline projects/vdk-plugins/airflow-provider-vdk/vdk_provider/sensors/vdk.py:11]: Can we pass workflow_id?

Comment (gageorgiev) [reply projects/vdk-plugins/airflow-provider-vdk/vdk_provider/sensors/vdk.py:11]: The workflow_id can be extracted through the context object in the poke method.

**Review by mivanov1988 — APPROVED** [review]

**Review by doks5 — APPROVED** [review]

**Pull request merged**